frontend_tests/node_tests/voting_widget.js

```diff
@@ -76,4 +84,291 @@ run_test('poll_data_holder my question', () => {
         question: 'best plan?',
     });
 
+    const invalid_vote_event = {
+        type: 'vote',
+        key: '98,1',
+        vote: 1,
+    };
+
+    blueslip.error = (msg) => {
+        assert.equal(msg, `unknown key for poll: ${invalid_vote_event.key}`);
+    };
+    data_holder.handle_event(sender_id, invalid_vote_event);
+    data = data_holder.get_widget_data();
+
+    const comment_outbound_event = data_holder.handle.new_comment.outbound('new comment');
+    assert.deepEqual(comment_outbound_event, {
+        type: 'new_comment',
+        idx: 2,
+        comment: 'new comment',
+    });
+
+    const new_question = 'Any new plan?';
+    const question_outbound_event = data_holder.handle.question.outbound(new_question);
+    assert.deepEqual(question_outbound_event, {
+        type: 'question',
+        question: new_question,
+    });
+
+    const vote_outbound_event = data_holder.handle.vote.outbound('99,1');
+    assert.deepEqual(vote_outbound_event, { type: 'vote', key: '99,1', vote: -1 });
+
+    vote_event = {
+        type: 'vote',
+        key: '99,1',
+        vote: -1,
+    };
+
+    data_holder.handle_event(sender_id, vote_event);
+    data = data_holder.get_widget_data();
+
+    assert.deepEqual(data, {
+        comments: [
+            {
+                comment: 'release now',
+                names: '',
+                count: 0,
+                key: '99,1',
+            },
+        ],
+        question: 'best plan?',
+    });
+});
+
+run_test('activate another person poll', () => {
+    people.is_my_user_id = return_false;
+    templates.render = (template_name) => {
+        if (template_name === 'poll-widget') {
+            return 'poll-widget';
+        } else if (template_name === 'poll-widget-results') {
```

Comment:
For this kind of thing, where each branch returns, I tend to prefer a bunch of "if" statements just to be more concise.  It's subjective, though.

Reply:
Okay.. I also prefer what you're referring here, but somehow I missed this while iterating.